coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -57,6 +58,111 @@ public async Task<Product> GetProductAsync(int productId)
             return product;
         }
 
+        public Task<InitiateProductResponse> AddProduct(InitiateProductRequest newProduct, IEnumerable<UserGroup> allowedUserGroups)
+        {
+            var product = new Product()
+            {
+                Price = newProduct.Price,
+                Description = newProduct.Description,
+                Name = newProduct.Name,
+                NumberOfTickets = newProduct.NumberOfTickets,
+                Visible = newProduct.Visible
+            };
+            
+            var productUserGroups = allowedUserGroups.Select(userGroup => new ProductUserGroup
+            {
+                ProductId = product.Id,
+                UserGroup = userGroup
+            }).ToList();
+
+            _context.ProductUserGroups.AddRange(productUserGroups);
+            
+            _context.Products.Add(product);
+            
+            _context.SaveChanges();
+
+
+            var result = new InitiateProductResponse
+            {
+                Price = product.Price,
+                Description = product.Description,
+                Name = product.Name,
+                NumberOfTickets = product.NumberOfTickets,
+                Visible = product.Visible
+            };
+
+            return Task.FromResult(result);
```

Comment:
Either remove Task from the return type to make it sequential, or make use of async methods inside it